iris/src/iris/pipeline/tutor_suggestion_pipeline.py

```diff
@@ -91,49 +93,131 @@ def __call__(self, dto: CommunicationTutorSuggestionPipelineExecutionDTO):
+        summary_pipeline = TutorSuggestionSummaryPipeline(
+            callback=self.callback, variant=self.variant
+        )
         try:
-            post_summary = summary.get("summary")
-        except AttributeError:
-            post_summary = None
+            summary = summary_pipeline(dto=dto)
+        except AttributeError as e:
+            logger.error("AttributeError in summary pipeline: %s", str(e))
+            self.callback.error("Error running summary pipeline")
+            return
+        except Exception as e:
+            logger.error("Unexpected error in summary pipeline: %s", str(e))
+            self.callback.error("Unexpected error running summary pipeline")
+            return
+
+        logger.info(summary)
+
+        if summary is None:
+            self.callback.error("No summary was generated")
+            return
+
+        try:
+            is_question_str = summary.get("is_question", "").lower()
+            is_question = is_question_str in ["yes", "true", "1"]
+            number_of_answers = summary.get("num_answers")
+            summary = summary.get("summary")
+            logger.info(
+                "is_question: %s, num_answers: %s", is_question, number_of_answers
+            )
+        except (AttributeError, TypeError) as e:
+            logger.error("Error parsing summary JSON: %s", str(e))
+            self.callback.error("Error parsing summary JSON")
+            return
+
+        if is_question and number_of_answers > 0:
+            self.callback.in_progress("Checking if questions is already answered")
+
+            prompt = ChatPromptTemplate.from_messages(
+                [("system", question_answered_prompt())]
+            )
+
+            try:
+                response = (prompt | self.pipeline).invoke({"thread_summary": summary})
+                self._append_tokens(
+                    self.llm.tokens, PipelineEnum.IRIS_TUTOR_SUGGESTION_PIPELINE
+                )
+            except Exception as e:
+                logger.error("Error checking if question is answered: %s", str(e))
+                response = "no"
+            if "yes" in response.lower():
+                self.callback.done(
+                    "The question has already been answered",
+                    artifact="The question has already been answered in the thread and should be marked as resolved.",
+                    tokens=self.tokens,
+                )
+                return
+
+        channel_type = get_channel_type(dto)
+
+        logging.info(channel_type)
+        if channel_type == "text_exercise":
+            self._run_text_exercise_pipeline(
+                text_exercise_dto=dto.text_exercise, summary=summary
+            )
+        elif channel_type == "programming_exercise":
+            self._run_programming_exercise_pipeline(dto=dto, summary=summary)
+        elif channel_type == "lecture":
+            self.callback.error("Not implemented yet")
+        else:
```

Comment:
This is currently redundant.
```suggestion
        else:
```